providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/FlagdGrpcInterceptor.java

```diff
@@ -0,0 +1,56 @@
+package dev.openfeature.contrib.providers.flagd;
+
+import io.grpc.CallOptions;
+import io.grpc.Channel;
+import io.grpc.ClientCall;
+import io.grpc.ClientInterceptor;
+import io.grpc.ForwardingClientCall;
+import io.grpc.Metadata;
+import io.grpc.MethodDescriptor;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.propagation.TextMapSetter;
+import io.opentelemetry.sdk.OpenTelemetrySdk;
```

Comment:
OpenTelemetry recommends only relying on the API for 3rd party libraries. Only App developer should use an implementation of the SDK. This way, you won't get into the case where different SDKs are loaded into the runtime, creating a mess in handling the context.

Reply:
This is fixed with the latest commits and migrated to `OpenTelemetry`